src/modules/Channel/components/MessageInput/useHandleUploadFiles.ts

```diff
@@ -0,0 +1,84 @@
+import { useCallback } from 'react';
+import { Logger } from '../../../../lib/SendbirdState';
+import { SendMFMFunctionType } from '../../context/hooks/useSendMultipleFilesMessage';
+import useSendbirdStateContext from '../../../../hooks/useSendbirdStateContext';
+import { SendableMessageType, isImage } from '../../../../utils';
+
+/**
+ * The handleUploadFiles is a function sending a FileMessage and MultipleFilesMessage
+ * by the received FileList from the ChangeEvent of MessageInput component.
+ */
+
+const FILE_SIZE_LIMIT = 300000000; // 300MB
+
+interface UseHandleUploadFilesDynamicProps {
+  sendFileMessage: (file: File, quoteMessage?: SendableMessageType) => void;
+  sendMultipleFilesMessage: SendMFMFunctionType;
+  quoteMessage?: SendableMessageType;
+}
+interface UseHandleUploadFilesStaticProps {
+  logger: Logger;
+}
+export type HandleUploadFunctionType = (files: FileList) => void;
+
+export const useHandleUploadFiles = ({
+  sendFileMessage,
+  sendMultipleFilesMessage,
+  quoteMessage,
+}: UseHandleUploadFilesDynamicProps, {
+  logger,
+}: UseHandleUploadFilesStaticProps): Array<HandleUploadFunctionType> => {
+  const { config } = useSendbirdStateContext();
+  const uikitMultipleFilesMessageLimit = config?.uikitMultipleFilesMessageLimit ?? 15;
+
+  const handleUploadeFiles = useCallback((fileList: FileList) => {
+    const files: File[] = Array.from(fileList);
+
+    if (!sendFileMessage || !sendMultipleFilesMessage) {
+      logger.warning('Channel|useHandleUploadFiles: required functions are undefined', { sendFileMessage, sendMultipleFilesMessage });
+      return;
+    }
+    if (files.length > 0) {
```

Comment:
Small suggestion; using ts-pattern in here would be better in terms of readability. 
```
import { match } from 'ts-pattern'
match(files.length)
  // put any child if-else inside of when stmt 
  .when(n => n  > 0, () => {...})
  .when(n => n  === 1, () => {...})
  .when(n => n  > 1, () => {...})
```